File: Examples/CSharp/Working-with-Objects/ConvertBoxMeshtoTriangleMeshCustomMemoryLayout.cs
using System;
using System.IO;
using System.Collections;
using Aspose.ThreeD;
using Aspose.ThreeD.Animation;
using Aspose.ThreeD.Entities;
using Aspose.ThreeD.Formats;
using Aspose.ThreeD.Utilities;
using System.Runtime.InteropServices;

namespace Aspose._3D.Examples.CSharp._Working_with_Objects
{
    class ConvertBoxMeshtoTriangleMeshCustomMemoryLayout
    {
        public static void Run()
        {
            // Initialize scene object
            Scene scene = new Scene();

            // Initialize Node class object
            Node cubeNode = new Node("box");

            //ExStart:ConvertBoxMeshtoTriangleMeshCustomMemoryLayout
            // get mesh of the Box
            Mesh box = (new Box()).ToMesh();
            //create a customized vertex layout
            VertexDeclaration vd = new VertexDeclaration();
            VertexField position = vd.AddField(VertexFieldDataType.FVector4, VertexFieldSemantic.Position);
            vd.AddField(VertexFieldDataType.FVector3, VertexFieldSemantic.Normal);
            // get a triangle mesh
            TriMesh triMesh = TriMesh.FromMesh(box);
            //ExEnd:ConvertBoxMeshtoTriangleMeshCustomMemoryLayout

            // Point node to the Mesh geometry
            cubeNode.Entity = box;

            // Add Node to a scene
            scene.RootNode.ChildNodes.Add(cubeNode);

            // The path to the documents directory.
            string MyDir = RunExamples.GetDataDir() + RunExamples.GetOutputFilePath("BoxToTriangleMeshCustomMemoryLayoutScene.fbx");

            // Save 3D scene in the supported file formats
            scene.Save(MyDir, FileFormat.FBX7400ASCII);

            Console.WriteLine("\n Converted a Box mesh to triangle mesh with custom memory layout of the vertex successfully.\nFile saved at " + MyDir);
        }
    }
}

File: Examples/VisualBasic/Working-with-Objects/ConvertBoxMeshtoTriangleMeshCustomMemoryLayout.vb
Imports System.IO
Imports System.Collections
Imports Aspose.ThreeD
Imports Aspose.ThreeD.Animation
Imports Aspose.ThreeD.Entities
Imports Aspose.ThreeD.Formats
Imports Aspose.ThreeD.Utilities
Imports System.Runtime.InteropServices

Namespace Working_with_Objects
    Class ConvertBoxMeshtoTriangleMeshCustomMemoryLayout
        Public Shared Sub Run()
            ' Initialize scene object
            Dim scene As New Scene()

            ' Initialize Node class object
            Dim cubeNode As New Node("box")

            ' ExStart:ConvertBoxMeshtoTriangleMeshCustomMemoryLayout
            ' Get mesh of the Box
            Dim box As Mesh = (New Box()).ToMesh()
            ' Create a customized vertex layout
            Dim vd As New VertexDeclaration()
            Dim position As VertexField = vd.AddField(VertexFieldDataType.FVector4, VertexFieldSemantic.Position)
            vd.AddField(VertexFieldDataType.FVector3, VertexFieldSemantic.Normal)
            ' Get a triangle mesh
            Dim triMesh__1 As TriMesh = TriMesh.FromMesh(box)
            ' ExEnd:ConvertBoxMeshtoTriangleMeshCustomMemoryLayout

            ' Point node to the Mesh geometry
            cubeNode.Entity = box

            ' Add Node to a scene
            scene.RootNode.ChildNodes.Add(cubeNode)

            ' The path to the documents directory.
            Dim MyDir As String = RunExamples.GetDataDir() + RunExamples.GetOutputFilePath("BoxToTriangleMeshCustomMemoryLayoutScene.fbx")

            ' Save 3D scene in the supported file formats
            scene.Save(MyDir, FileFormat.FBX7400ASCII)

            Console.WriteLine(Convert.ToString(vbLf & " Converted a Box mesh to triangle mesh with custom memory layout of the vertex successfully." & vbLf & "File saved at ") & MyDir)
        End Sub
    End Class
End Namespace